scipy-stubs/signal/_peak_finding.pyi

```diff
@@ -1,37 +1,72 @@
+    left_ips: NotRequired[npt.NDArray[np.float64]]
+    right_ips: NotRequired[npt.NDArray[np.float64]]
+    plateau_sizes: NotRequired[npt.NDArray[np.intp]]
+    left_edges: NotRequired[npt.NDArray[np.intp]]
+    right_edges: NotRequired[npt.NDArray[np.intp]]
+
+def argrelmin(
+    data: npt.NDArray[np.generic], axis: int = 0, order: int = 1, mode: _Mode = "clip"
+) -> tuple[npt.NDArray[np.intp], ...]: ...
+def argrelmax(
+    data: npt.NDArray[np.generic], axis: int = 0, order: int = 1, mode: _Mode = "clip"
+) -> tuple[npt.NDArray[np.intp], ...]: ...
+def argrelextrema(
+    data: npt.NDArray[np.generic],
+    comparator: Callable[[npt.NDArray[np.generic], npt.NDArray[np.generic]], npt.NDArray[np.bool_]],
+    axis: int = 0,
+    order: int = 1,
+    mode: _Mode = "clip",
+) -> tuple[npt.NDArray[np.intp], ...]: ...
+def peak_prominences(x: npt.ArrayLike, peaks: npt.ArrayLike, wlen: int | None = None) -> _ProminencesResult: ...
 def peak_widths(
-    x: Untyped,
-    peaks: Untyped,
+    x: npt.ArrayLike,
+    peaks: npt.ArrayLike,
     rel_height: float = 0.5,
-    prominence_data: Untyped | None = None,
-    wlen: Untyped | None = None,
-) -> Untyped: ...
+    prominence_data: _ProminencesResult | None = None,
+    wlen: int | None = None,
+) -> _WidthsResult: ...
 def find_peaks(
-    x: Untyped,
-    height: Untyped | None = None,
-    threshold: Untyped | None = None,
-    distance: Untyped | None = None,
-    prominence: Untyped | None = None,
-    width: Untyped | None = None,
-    wlen: Untyped | None = None,
+    x: npt.ArrayLike,
+    height: float | npt.NDArray[np.float64] | tuple[float | None, float | None] | None = None,
+    threshold: float | npt.NDArray[np.float64] | tuple[float | None, float | None] | None = None,
+    distance: np.float64 | None = None,
+    prominence: float | npt.NDArray[np.float64] | tuple[float | None, float | None] | None = None,
+    width: float | npt.NDArray[np.float64] | tuple[float | None, float | None] | None = None,
+    wlen: int | None = None,
     rel_height: float = 0.5,
-    plateau_size: Untyped | None = None,
-) -> Untyped: ...
+    plateau_size: int | npt.NDArray[np.intp] | tuple[int | None, int | None] | None = None,
+) -> tuple[npt.NDArray[np.intp], _FindPeaksResultsDict]: ...
 def find_peaks_cwt(
-    vector: Untyped,
-    widths: Untyped,
-    wavelet: Untyped | None = None,
-    max_distances: Untyped | None = None,
-    gap_thresh: Untyped | None = None,
-    min_length: Untyped | None = None,
+    vector: npt.ArrayLike,
+    widths: npt.ArrayLike,
+    wavelet: Callable[Concatenate[int, float, ...], npt.NDArray[np.float64]] | None = None,
+    max_distances: Sequence[int] | None = None,
+    gap_thresh: float | None = None,
+    min_length: int | None = None,
     min_snr: int = 1,
     noise_perc: int = 10,
-    window_size: Untyped | None = None,
-) -> Untyped: ...
+    window_size: int | None = None,
+) -> npt.NDArray[np.intp]: ...
```

Comment:
If you're feeling up to it, you could narrow this to a 1-D array as e.g. `optype.numpy.Array[tuple[int], np.intp]` (but it's also perfectly fine if you leave it like this).